tests/integrations/tso/client_test.go

```diff
@@ -197,6 +197,32 @@ func (suite *tsoClientTestSuite) TestGetTSAsync() {
 	wg.Wait()
 }
 
+func (suite *tsoClientTestSuite) TestDiscoverTSOServiceWithLegacyPath() {
+	re := suite.Require()
+	var wg sync.WaitGroup
+	wg.Add(tsoRequestConcurrencyNumber)
+	for i := 0; i < tsoRequestConcurrencyNumber; i++ {
+		go func() {
+			defer wg.Done()
+			// Simulate the case that the server has lower version than the client and returns no tso addrs
+			// in the GetClusterInfo RPC.
+			re.NoError(failpoint.Enable("github.com/tikv/pd/client/serverReturnsNoTSOAddrs", "return(true)"))
+			defer re.NoError(failpoint.Disable("github.com/tikv/pd/client/serverReturnsNoTSOAddrs"))
```

Comment:
No need to enable the failpoint in each goroutine?

Reply:
need to use 
```
defer func() {
      re.NoError(failpoint.Disable("github.com/tikv/pd/client/serverReturnsNoTSOAddrs"))
}
```
or disable it in the end of the goroutine

Reply:
figured out. failpoint isn't goroutine isolated.